### PR TITLE
feat(hatchery/openstack): check flavor

### PR DESCRIPTION
**Pull request by yesnault**

### CODE REVIEW

File: engine/hatchery/openstack/init.go [hunk 1]
```diff
@@ -58,11 +58,11 @@ func (h *HatcheryOpenstack) InitHatchery(ctx context.Context) error {
 func (h *HatcheryOpenstack) initFlavors() error {
 	all, err := flavors.ListDetail(h.openstackClient, nil).AllPages()
 	if err != nil {
-		return fmt.Errorf("initFlavors> error on flavors.ListDetail: %s", err)
+		return fmt.Errorf("initFlavors> error on flavors.ListDetail: %v", err)
```
Comment (fsamin) [inline engine/hatchery/openstack/init.go:61]:
```suggestion
		return sdk.WithStack(fmt.Errorf("initFlavors> error on flavors.ListDetail: %v", err)
```)

File: engine/hatchery/openstack/init.go [hunk 2]
```diff
@@ -58,11 +58,11 @@ func (h *HatcheryOpenstack) InitHatchery(ctx context.Context) error {
...
 	}
 	lflavors, err := flavors.ExtractFlavors(all)
 	if err != nil {
-		return fmt.Errorf("initFlavors> error on flavors.ExtractFlavors: %s", err)
+		return fmt.Errorf("initFlavors> error on flavors.ExtractFlavors: %v", err)
```
Comment (fsamin) [inline engine/hatchery/openstack/init.go:65]:
```suggestion
		return sdk.WithStack(fmt.Errorf("initFlavors> error on flavors.ExtractFlavors: %v", err))
```

File: engine/hatchery/openstack/init.go [hunk 3]
```diff
@@ -71,11 +71,11 @@ func (h *HatcheryOpenstack) initFlavors() error {
 func (h *HatcheryOpenstack) initNetworks() error {
 	all, err := tenantnetworks.List(h.openstackClient).AllPages()
 	if err != nil {
-		return fmt.Errorf("initNetworks> Unable to get Network: %s", err)
+		return fmt.Errorf("initNetworks> Unable to get Network: %v", err)
```
Comment (fsamin) [inline engine/hatchery/openstack/init.go:74]:
```suggestion
		return sdk.WithStack(fmt.Errorf("initNetworks> Unable to get Network: %v", err))
```

File: engine/hatchery/openstack/init.go [hunk 4]
```diff
@@ -71,11 +71,11 @@ func (h *HatcheryOpenstack) initFlavors() error {
...
 	}
 	nets, err := tenantnetworks.ExtractNetworks(all)
 	if err != nil {
-		return fmt.Errorf("initNetworks> Unable to get Network: %s", err)
+		return fmt.Errorf("initNetworks> Unable to get Network: %v", err)
```
Comment (fsamin) [inline engine/hatchery/openstack/init.go:78]:
```suggestion
		return sdk.WithStack(fmt.Errorf("initNetworks> Unable to get Network: %v", err))
```

File: engine/hatchery/openstack/openstack.go [hunk 5]
```diff
@@ -157,6 +157,15 @@ func (h *HatcheryOpenstack) WorkerModelsEnabled() ([]sdk.Model, error) {
 // CanSpawn return wether or not hatchery can spawn model
 // requirements are not supported
 func (h *HatcheryOpenstack) CanSpawn(ctx context.Context, model *sdk.Model, jobID int64, requirements []sdk.Requirement) bool {
+	// if there is a model, we have to check if the flavor attached to model is knowned by this hatchery
+	if model != nil {
+		if _, err := h.flavorID(model.ModelVirtualMachine.Flavor); err != nil {
+			log.Debug("CanSpawn> flavor '%s' not found", model.ModelVirtualMachine.Flavor)
```
Comment (fsamin) [inline engine/hatchery/openstack/openstack.go:163]: it could be nice to check the error

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#13083.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✘

File: engine/hatchery/openstack/openstack.go (hunk 5 above)
Comment (yesnault) [reply engine/hatchery/openstack/openstack.go]: fixed

Comment (fsamin) [reply engine/hatchery/openstack/openstack.go]: sorry, i don't see how you are checking the error is "flavor not found"

**Pull request merged**